### PR TITLE
Recreate assembled model JAR for Flink if it got removed (e.g. by systemd-tmpfiles)

### PR DESCRIPTION
**Pull request by piotrp**
## Describe your changes

Recreate model JAR for Flink if model written to temp directory got removed. E.g. systemd-tmpfiles on CentOS removes files from default `java.io.tmpdir` after 10 days.

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkModelJar.scala [hunk 1]
```diff
@@ -3,31 +3,30 @@ package pl.touk.nussknacker.engine.management
...
-class FlinkModelJar extends LazyLogging {
+class FlinkModelJar(modelUrls: List[URL], includeDropwizardLibsImplicitly: Boolean = true) extends LazyLogging {
 
-  // TODO: handle multiple models?
   private var modelFile: Option[File] = None
 
   // we want to have *different* file names for *different* model data (e.g. after rebuild etc.)
   // currently we just generate random file names
-  def buildJobJar(modelData: BaseModelData, includeDropwizardLibsImplicitly: Boolean = true): File = synchronized {
+  def buildJobJar(): File = synchronized {
```
Comment (arkadius) [inline engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkModelJar.scala:18]: I think that it's time to change the name of this method. It doesn't build jar every time when it is invoked. Also I would change the name of the class. It is a kind of `FlinkModeljarProvider`

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala [hunk 2]
```diff
@@ -26,7 +25,7 @@ class FlinkRestManager(config: FlinkConfig, modelData: BaseModelData, mainClassN
 ) extends FlinkDeploymentManager(modelData, config.shouldVerifyBeforeDeploy, mainClassName)
     with LazyLogging {
 
-  protected lazy val jarFile: File = new FlinkModelJar().buildJobJar(modelData)
+  private val modelJar = new FlinkModelJar(modelData.modelClassLoaderUrls)
```
Comment (arkadius) [inline engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala:28]: `modelJarPovider`?

**Review by arkadius — APPROVED** [review]

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala (hunk 2 above)
Comment (piotrp) [reply engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala]: Done

File: engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkModelJar.scala (hunk 1 above)
Comment (piotrp) [reply engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkModelJar.scala]: Done

**Review by arkadius — APPROVED** [review]

**Pull request merged**